patterns/ecs-v1/linux-syslog

```diff
@@ -1,16 +1,16 @@
 SYSLOG5424PRINTASCII [!-~]+
 
-SYSLOGBASE2 (?:%{SYSLOGTIMESTAMP:timestamp}|%{TIMESTAMP_ISO8601:timestamp8601}) (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:logsource}+(?: %{SYSLOGPROG}:|)
-SYSLOGPAMSESSION %{SYSLOGBASE} (?=%{GREEDYDATA:message})%{WORD:pam_module}\(%{DATA:pam_caller}\): session %{WORD:pam_session_state} for user %{USERNAME:username}(?: by %{GREEDYDATA:pam_by})?
+SYSLOGBASE2 (?:%{SYSLOGTIMESTAMP:timestamp}|%{TIMESTAMP_ISO8601:timestamp8601}) (?:%{SYSLOGFACILITY} )?%{SYSLOGHOST:event.provider}+(?: %{SYSLOGPROG}:|)
+SYSLOGPAMSESSION %{SYSLOGBASE} (?=%{GREEDYDATA:message})%{WORD:log.syslog.pam.module}\(%{DATA:log.syslog.pam.caller}\): session %{WORD:log.syslog.pam.session_state} for user %{USERNAME:log.syslog.pam.user}(?: by %{GREEDYDATA})?
 
 CRON_ACTION [A-Z ]+
 CRONLOG %{SYSLOGBASE} \(%{USER:user}\) %{CRON_ACTION:action} \(%{DATA:message}\)
 
 SYSLOGLINE %{SYSLOGBASE2} %{GREEDYDATA:message}
 
 # IETF 5424 syslog(8) format (see http://www.rfc-editor.org/info/rfc5424)
-SYSLOG5424PRI <%{NONNEGINT:syslog5424_pri}>
+SYSLOG5424PRI <%{NONNEGINT:log.syslog.facility.code}>
 SYSLOG5424SD \[%{DATA}\]+
-SYSLOG5424BASE %{SYSLOG5424PRI}%{NONNEGINT:syslog5424_ver} +(?:%{TIMESTAMP_ISO8601:syslog5424_ts}|-) +(?:%{IPORHOST:syslog5424_host}|-) +(-|%{SYSLOG5424PRINTASCII:syslog5424_app}) +(-|%{SYSLOG5424PRINTASCII:syslog5424_proc}) +(-|%{SYSLOG5424PRINTASCII:syslog5424_msgid}) +(?:%{SYSLOG5424SD:syslog5424_sd}|-|)
+SYSLOG5424BASE %{SYSLOG5424PRI}%{NONNEGINT:log.syslog.version} +(?:%{TIMESTAMP_ISO8601:timestamp}|-) +(?:%{IPORHOST:event.provider}|-) +(-|%{SYSLOG5424PRINTASCII:log.syslog.app_name}) +(-|%{SYSLOG5424PRINTASCII:log.syslog.procid}) +(-|%{SYSLOG5424PRINTASCII:log.syslog.msgid}) +(?:%{SYSLOG5424SD:log.syslog.structured_data}|-|)
```

Comment:
these should capture the same fields as the SYSLOGLINE (unlike previously)
\+ the standardized new pieces: `log.syslog.app_name`, `log.syslog.procid` etc.

Reply:
Do these need to be logstash-style field references, or does Grok work magic to handle the nesting when given dot-style nesting?

Reply:
grok doesn't do any magic so they're just flat for now (with the dots).
until all names are reviewed I am going with dot notation - for sanity :wink: 
these will need to be renamed on a final review `log.syslog.version` -> `[log][syslog][version]`

Reply:
these are going to be `system.syslog` prefixed e.g. `system.syslog.app_name` 
(for beats compatibility - despite beats not matching any of these)

Reply:
seems I got confused here, there's a new discussion on these fields so I will close this one